src/helpers.js

```diff
@@ -107,9 +107,18 @@ function checkIfConditionMatches(node, formValues, formFields) {
       return currentProperty.enum.includes(value);
     }
 
-    const { inputType } = getField(name, formFields);
-
-    return validateFieldSchema({ ...currentProperty, inputType, required: true }, value);
+    const field = getField(name, formFields);
+
+    return validateFieldSchema(
+      {
+        options: field.options,
```

Comment:
note: This solves a bug where conditionals with select fields weren't working. For example, the JSF would break on the following JSON Schema because it wouldn't be able to process the `if.properties.browser` correctly.

```json
{
  "additionalProperties": false,
  "allOf": [
    {
      "if": {
        "properties": {
          "browser": {
            "pattern": "."
          }
        },
        "required": [
          "browser"
        ]
      },
      "then": {},
      "else": {
        "properties": {
          "model": false
        }
      }
    }
  ],
  "properties": {
    "browser": {
      "oneOf": [
        {
          "const": "chr",
          "title": "Chrome"
        },
        {
          "const": "ff",
          "title": "Firefox"
        }
      ],
      "title": "Browser",
      "type": "string",
      "x-jsf-presentation": {
        "inputType": "select"
      }
    },
    "model": {
      "maxLength": 1000,
      "title": "Browser version",
      "type": "string",
      "x-jsf-presentation": {
        "inputType": "textarea"
      }
    }
  }
}
```

<img width="1661" alt="image" src="https://github.com/remoteoss/json-schema-form/assets/14869087/9e6d602f-32c5-4f02-91fb-b1df273f1e75">